### DISCUSSION TITLE
Search Box is not working with search result

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2716] [Q&A]
I am using the latest version of PnP-modern-search.
I have added 3 web parts; PnP Search Filter + PnP Search Box + PnP Search Result.

Now i connect the PnP Search Result with the PnP Search Box as follow:-

![image](https://user-images.githubusercontent.com/79726511/214740432-177b04c0-4848-46a3-b82f-31d1952d1bc2.png)

but currently when i do a free text search i will get all results (29 results instead of 1 result only), as follow:-

![image](https://user-images.githubusercontent.com/79726511/214740666-63ac335a-5907-46a2-8a46-69dd582b5e4a.png)

any advice please?

**Comment by kasperbolarsen** [comment]
That sounds like an error in the Results web parts Query template. Does it contains {searchTerms}  ?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  yes it contain the {searchTerms}

**Reply by kasperbolarsen** [reply]
And nothing else?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  nothing else inside the Query Template it only contains this 

![image](https://user-images.githubusercontent.com/79726511/214834031-b5e4a3a0-81a0-494e-8d2f-c96cb3013dbb.png)

**Comment by wobba** [comment]
Can you go over the tutorials at https://microsoft-search.github.io/pnp-modern-search/scenarios/ and see if you resolve your issue?

**Reply by mvcsharepointdev** [reply]
@wobba i already went through it , but unfortunelty it is still not working.. could this be a bug in the latest version ? as previously this scenario use to work well for me

**Reply by kasperbolarsen** [reply]
Have you run the same query in either SP Editor/ SP Query Tool or code ?

**Reply by kasperbolarsen** [reply]
@mvcsharepointdev the reason why I ask you to test the query is to eliminate a gotcha like some other property than the title contains your search term 😉

**Reply by wobba** [reply]
You can also look at the request JSON and response JSON on the API call and look for issues. Or use the debug template and look for the queryModification response which tells you the full executed query.
